File: src/main/java/net/starlegacy/feature/gas/collectionfactors/SpaceOnlyFactor.java
package net.starlegacy.feature.gas.collectionfactors;

import net.starlegacy.feature.space.SpaceWorlds;
import org.bukkit.Location;

public class SpaceOnlyFactor extends CollectionFactor {

    @Override
    public boolean factor(Location location) {
        return SpaceWorlds.INSTANCE.contains(location.getWorld());
    }
}

File: server/src/main/kotlin/net/horizonsend/ion/server/features/gas/collectionfactors/SpaceOnlyFactor.kt
package net.horizonsend.ion.server.features.gas.collectionfactors

import net.horizonsend.ion.server.features.world.IonWorld.Companion.ion
import net.horizonsend.ion.server.features.world.WorldFlag
import org.bukkit.Location

class SpaceOnlyFactor : CollectionFactor() {
    override fun factor(location: Location): Boolean {
        return location.world.ion.hasFlag(WorldFlag.SPACE_WORLD)
    }

	override fun canBeFound(location: Location): Boolean = factor(location)
}
